### PR TITLE
fix: adding throttled status code for server unavailable error in salesforce

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

For salesforce when the salesforce server is busy or unavailable we were throwing retriable which was being failed eventually by retrying  3 times.
Now, we are throwing 429 status code appropriately.

## What is the related Linear task?

Resolves INT-2847

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/destinations/salesforce/utils.js [hunk 1]
```diff
@@ -23,6 +23,13 @@ const {
 
 const ACCESS_TOKEN_CACHE = new Cache(ACCESS_TOKEN_CACHE_TTL);
 
+const getErrorMessage = (response) => {
```
Comment (utsabc) [inline src/v0/destinations/salesforce/utils.js:26]: Lets document the format of error messages here with reference

File: src/v0/destinations/salesforce/utils.js [hunk 2]
```diff
@@ -77,15 +84,18 @@ const salesforceResponseHandler = (destResponse, sourceMessage, authKey, authori
...
-      throw new RetryableError(
-        `${DESTINATION} Request Failed - due to "${
-          response && Array.isArray(response) && response[0]?.message?.length > 0
-            ? response[0].message
-            : JSON.stringify(response)
-        }", (Retryable) ${sourceMessage}`,
-        500,
-        destResponse,
-      );
+      if (matchErrorCode('SERVER_UNAVAILABLE')) {
```
Comment (utsabc) [inline src/v0/destinations/salesforce/utils.js:87]: Lets document the reference of this error message

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3862/test-report.html

File: src/v0/destinations/salesforce/utils.js (hunk 1 above)
Comment (shrouti1507) [reply src/v0/destinations/salesforce/utils.js:42]: done

**Review by utsabc — APPROVED** [review]

File: src/v0/destinations/salesforce/utils.js [hunk 3]
```diff
@@ -77,15 +100,19 @@ const salesforceResponseHandler = (destResponse, sourceMessage, authKey, authori
...
-            ? response[0].message
-            : JSON.stringify(response)
-        }", (Retryable) ${sourceMessage}`,
-        500,
-        destResponse,
-      );
+      // ref : https://help.salesforce.com/s/articleView?id=000387190&type=1
+      if (matchErrorCode('SERVER_UNAVAILABLE')) {
+        throw new ThrottledError(
+          `${DESTINATION} Request Failed: ${status} - due to Search unavailable, ${sourceMessage}`,
```
Comment (krishna2020) [inline src/v0/destinations/salesforce/utils.js:106]: Can't we use `getErrorMessage` here too - helps if there are any other error types that can come under SERVER_UNAVAILABLE code

Comment (shrouti1507) [reply src/v0/destinations/salesforce/utils.js]: edited

**Review by utsabc — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**